### ISSUE TITLE
showInAppNotification does not dismiss on Android

### ISSUE THREAD

**Issue by rodrigopk** [wix/react-native-navigation#3613]
### Issue Description
Whenever I add a notification using the `showInAppNotification` method, the notification is not dismissed on swipe as expected.

### Steps to Reproduce / Code Snippets / Screenshots
1 - Add a notification using the `showInAppNotification` method:
```
this.props.navigator.showInAppNotification({
    screen: "example.InAppNotification",
    passProps: { ...notificationProps },
    autoDismissTimerSec: 5,
  });
```
2 - Run the application on an Android device;
3 - Trigger the notification display;
4 - Try to swipe to notification up to dismiss

---
### Environment
* React Native Navigation version: 1.1.433
* React Native version: 0.55.4
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?):  Device - SM-G950F - 8.0.0 - Debug

**Comment by 1453009** [comment]
Is it support swipe to dismiss? (pls tell me)
I implemented my own dismiss bar using scrollview and animate opacity

**Comment by rodrigopk** [comment]
@1453009 Yes, I want to support swipe to dismiss on Android. On iOS the swipe is working out of the box. I liked your idea, I'll try to implement something based on that.

**Comment by 1453009** [comment]
@rodrigopk is it on document? can you show me, I'm using v1.1

**Comment by rodrigopk** [comment]
@1453009 I'm sorry, I didn't understand your question. Are you asking about the swipe to dismiss notification on iOS?
If so, I could not find any mention to it on the docs as well, but you can see it implemented on the method `initWithParams` of `RCCNotification.m`:

```
// react-native-navigation/blob/master/ios/RCCNotification.m

if ([params[@"dismissWithSwipe"] boolValue])
        {
            UISwipeGestureRecognizer *swipeGesture = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:@selector(performDismiss)];
            swipeGesture.direction = [self swipeDirection];
            [self addGestureRecognizer:swipeGesture];
        }
```
The default value for `dismissWithSwipe` is true on the iOS implementation, and you can see the behaviour by yourself with the example code I posted on the issue.

**Comment by rodrigopk** [comment]
Got the swipe dismiss on Android working using the `react-native-swipe-gestures` library and and `LayoutAnimation`, based on @1453009 idea.

Here's the general idea of the workaround:

```
class MyNotification extends React.PureComponent {

   state = {
      isNotificationDismissed: false,
  };

  render() {
      return (
        <GestureRecognizer
          style={this.getDisplayStyle()}
          onSwipeUp={this.handleSwipeUp}>
          {this.renderNotificationContent()}
        </GestureRecognizer>
      );
    }

    getDisplayStyle = () => {
      if (this.state.isNotificationDismissed) return styles.hiddenNotification;
      return styles.visibleNotification;
    };

    handleSwipeUp = () => {
      LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
      this.setState({ isNotificationDismissed: true });
    };

   . 
   .
   .
}

const styles = StyleSheet.create({
    visibleNotification: {
      height: 64,
    },
    hiddenNotification: {
      height: 0,
    },
});
```

**Comment by 1453009** [comment]
wow, this solution is cool
initially I tried panresponder but can't get any result then I use very basic way of scrollview horizontally and render null if reach some point and It works

But I'm still not sure if swipe gesture is originally supported on ios (I swiped left and right), is this line make it works?
`passProps: { ...notificationProps }`

**Comment by rodrigopk** [comment]
@1453009 The swipe direction to dismiss the notification on iOS is up, and as far as I could see there's no way of changing that through props

**Comment by datvp09** [comment]
![simulator_screen_shot_-_iphone_6_-_2018-06-18_at_09 48 10](https://user-images.githubusercontent.com/25218577/45256527-a8643680-b3c1-11e8-940a-b7520941d69d.png)

@rodrigopk my in app notification is like this, the direction is right (or left) to dismiss it. So if it's not originally supported horizontally, I will need to implement my own base on your idea above :(
